baseline_testing/run_baselines.py

```diff
@@ -0,0 +1,166 @@
+
+    Returns
+    -------
+    kpi (float): Key Performance Indicator (KPI) for the scenario.
+    df_res (pd.DataFrame): DataFrame containing simulation results.
+    """
+    control_module = 'examples.python.controllers.baseline'
+    step = length
+    kpi, df_res, _ , _ = control_test(control_module, start_time, warmup_period, length,
+                                                      scenario=scenario_params, step=step)
+    return kpi, df_res
+
+def save_kpi_result(scenario_params, kpi):
+    """
+    Save KPI result for a scenario to a JSON file.
+
+    Parameters
+    ----------
+    scenario_params (Scenario): Parameters for the scenario.
+    kpi (float): Key Performance Indicator (KPI) for the scenario.
+    """
+    script_dir = os.path.dirname(os.path.abspath(__file__))  # Directory of the current Python script
+    result_folder = os.path.join(script_dir, "result")  # Result folder path
+    os.makedirs(result_folder, exist_ok=True)  # Create result folder if it doesn't exist
+
+    file_path = os.path.join(result_folder, 'kpi_results.json')  # File path within the result folder
+
+    if os.path.exists(file_path):
+        # If the file exists, load existing data
+        with open(file_path, 'r') as f:
+            kpi_results = json.load(f)
+    else:
+        kpi_results = []
+
+    # Append new data
+    kpi_results.append({'scenario_params': scenario_params, 'kpi': kpi})
+
+    # Write back to the file
+    with open(file_path, 'w') as f:
+        json.dump(kpi_results, f)
+        
+
+def save_measurements_to_csv(df_res, scenario_name):
+    """
+    Save simulation results to a CSV file.
+
+    Parameters
+    ----------
+    df_res (pd.DataFrame): DataFrame containing simulation results.
+    scenario_name (str): Name of the scenario.
+    """
+
+    script_dir = os.path.dirname(os.path.abspath(__file__))  # Directory of the current Python script
+    result_folder = os.path.join(script_dir, "result")  # Result folder path
+    os.makedirs(result_folder, exist_ok=True)  # Create result folder if it doesn't exist
+
+    file_name = f'df_res_{scenario_name}.csv'  # File name
+    file_path = os.path.join(result_folder, file_name)  # File path within the result folder
+
+    df_res.to_csv(file_path, index=True)  # Save DataFrame to CSV file
+
+    
+def main(test_case):
+    """
+    Main function to run simulations based on the specified test case.
+
+    Parameters
+    ----------
+    test_case (str): Name of the test case.
+    """
+    # Load configuration from JSON file
+    with open('config.json') as config_file:
+        config = json.load(config_file)
+    
+    # Check if the test case is in the configuration
+    if test_case not in config:
+        print(f"Test case '{test_case}' not found in config.")
+        sys.exit(1)
+    
+    model_config = config[test_case]
+    
+    # Run user-defined baseline control testing or run all scenarios
+    if model_config.get("run_user_defined_test", False):
```

Comment:
Are you sure this is valid since model_config, defined in Line 137, I think only contains fields in the specific test case, which are electricity_price and time_period?  Is "run_user_defined_test" checked correctly?

Related, it would be good to include the script(s) and config(s) you used to generate all the two-week rolling simulations (if they're pretty concise), which I think probably used the "user_defined" feature and maybe automated creation of the necessary config files?  Were they tested with this version of run_baselines.py?

Reply:
During this round of revision, I've tested both the "user_defined" feature with "true" and "false" (running predefined scenarios) settings. Additionally, I've included a script  `baselines/csv/run_all_scenarios.py` for running two-week rolling simulations to reproduce the baseline testing results.